src/test/unit/app/fact/FactApi.ts

```diff
@@ -377,4 +378,105 @@ describe('FactApi', () => {
     const api = new FactApi(mockAxios, mockLogger);
     await expect(api.updateEmails('No Slug', [])).rejects.toEqual(mockError);
   });
+
+
+  test('Should return results from getCourtTypes request', async () => {
+    const results = {
+      data: [
+        { id: 1, name:"Magistrates' Court", code: 123},
+        { id: 2, name:'Tribunal Court', code: 456}
+
+      ]
+    };
+
+    const mockAxios = { get: async () => results } as any;
+    const mockLogger = {} as any;
+
+    const api = new FactApi(mockAxios, mockLogger);
+
+    await expect(api.getCourtTypes()).resolves.toEqual(results.data);
+  });
+
+  test('Should return results and log error from getCourtTypes request', async () => {
+    const mockAxios = { get: async () => {
+      throw mockError;
+    }} as never;
+
+    const mockLogger = {
+      error: (message: string) => message,
+      info: (message: string) => message
+    } as never;
+    const api = new FactApi(mockAxios, mockLogger);
+
+    await expect(api.getCourtTypes()).rejects.toEqual(mockError);
+  });
+
+  test('Should log error and reject promise for failed getCourtTypes request', async () => {
+    const mockAxios = { get: async () => {
+      throw mockError;
+    }} as any;
+
+    const spy = jest.spyOn(mockLogger, 'info');
+    const api = new FactApi(mockAxios, mockLogger);
+
+    await expect(api.getCourtTypes()).rejects.toBe(mockError);
+    await expect(spy).toBeCalled();
+  });
+
+  test('Should return results from getCourtCourtTypes request', async () => {
+    const results = {
+      data: [
+        { id: 1, name:"Magistrates' Court", code: 123},
+        { id: 2, name:'Tribunal Court', code: 456}
+
+      ]
+    };
+
+    const mockAxios = { get: async () => results } as any;
+    const mockLogger = {} as any;
+
+    const api = new FactApi(mockAxios, mockLogger);
+
+    await expect(api.getCourtCourtTypes('London')).resolves.toEqual(results.data);
+  });
+
+  test('Should return results and log error from getCourtCourtTypes request', async () => {
+    const mockAxios = { get: async () => {
+      throw mockError;
+    }} as never;
+
+    const mockLogger = {
+      error: (message: string) => message,
+      info: (message: string) => message
+    } as never;
+    const api = new FactApi(mockAxios, mockLogger);
+
+    await expect(api.getCourtCourtTypes('No slug')).rejects.toEqual(mockError);
+  });
+
+  test('Should log error and reject promise for failed getCourtCourtTypes request', async () => {
+    const mockAxios = { get: async () => {
+      throw mockError;
+    }} as any;
+
+    const spy = jest.spyOn(mockLogger, 'info');
+    const api = new FactApi(mockAxios, mockLogger);
+
+    await expect(api.getCourtCourtTypes('No Slug')).rejects.toBe(mockError);
+    await expect(spy).toBeCalled();
+  });
+
+
+  test('Should return results and log error from updateCourtCourtTypes request', async () => {
```

Comment:
Minor issue - the test description doesn't really describe that it's testing failure

Reply:
altered test description